dashboard/src/components/LazyDate.tsx

```diff
@@ -0,0 +1,46 @@
+'use client';
+import { useDictionary } from '@/contexts/DictionaryContextProvider';
+import { interpolateReactTemplate } from '@/utils/reactTemplateStrings';
+import { useEffect, useMemo, useState, ReactNode } from 'react';
+
+type LazyDateProps = {
+  date?: Date;
+  locales?: Intl.LocalesArgument;
+  options?: Intl.DateTimeFormatOptions;
+  formatted?:
+    | ((formattedDateString: string) => React.ReactNode)
+    | [(formattedDateString: string) => string, Record<string, React.ReactNode>];
```

Comment:
Suggestion: I would leave it as this:
```suggestion
  formatted?: (formattedDateString: string) => React.ReactNode;
```

We can always extend it later, if needed